modules/installation-guide/partials/con_workspace-exposure-strategies.adoc

```diff
@@ -34,13 +33,26 @@ When any of those single-host types is used, all workspaces are deployed to sub-
 
 This is convenient for TLS-secured {prod-short} servers because it is sufficient to have a single certificate for the {prod-short} server, which will cover all the workspace component deployments as well.
 
-This strategy limits the exposed components and user applications. Any absolute URL generated on the server side that points back to the server does not work. This is because the server is hidden behind a path-rewriting Ingress that hides the workspace and the component-specific URL prefix from the server.
+There are two types of endpoints, that are defined in the devfiles, exposures in single-host strategy. To configured it, use `++CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_DEVFILE__ENDPOINT__EXPOSURE++`. It is effective only in single-host server strategy..
+
+=== devfile endpoints: `single-host`
+
+`++CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_DEVFILE__ENDPOINT__EXPOSURE: 'single-host'++`
+
+This single-host configuration exposes the endpoints on subpaths, e.g.: `++https://<che-host>/serverihzmuqqc/go-cli-server-8080++`. This limits the exposed components and user applications. Any absolute URL generated on the server side that points back to the server does not work. This is because the server is hidden behind a path-rewriting Ingress that hides the workspace and the component-specific URL prefix from the server.
 
 For example, when the user accesses the hypothetical `[{prod-url}/component-prefix-djh3d/app/index.php]` URL, the application sees the request coming to `++https://internal-host/app/index.php++`. If the application used the host in the URL that it generates in its UI, it would not work because the internal host is different from the externally visible host. However, if the application used an absolute path as the URL (for the example above, this would be `/app/index.php`), such URL would still not work. This is because on the outside, such URL does not point to the application, because it is missing the component-specific prefix.
 
 Therefore, only applications that use relative URLs in their UI work with the single-host workspace exposure strategy.
 
+=== devfile endpoints: `multi-host`
+`++CHE_INFRA_KUBERNETES_SINGLEHOST_WORKSPACE_DEVFILE__ENDPOINT__EXPOSURE: 'multi-host'++`
 
+This single-host configuration exposes the endpoints on subdomains, e.g.: `++http://serverihzmuqqc-go-cli-server-8080.<che-host>++`. These endpoints will be exposed on unsecured HTTP port. We're using dedicated Ingress or Route for such endpoints, even with `gateway` single-host setup.
```

Comment:
This single-host configuration exposes the endpoints on subdomains, e.g.: `++http://serverihzmuqqc-go-cli-server-8080.<che-host>++`. These endpoints are be exposed on an unsecured HTTP port. A dedicated Ingress or Route is used for such endpoints, even with `gateway` single-host setup.

Reply:
fixed